Add Makefile, make test folder output local

jacopofar opened the pull request:
This adds a Makefile with the commands to automate build, run and deletion.
I also moved the output folder path to be relative to the project

rustprooflabs commented on Makefile:
```diff
@@ -0,0 +1,32 @@
+docker-clean:
+	# when docker runs as a demon the mounted folders have permissions and cannot be deleted by the current user
+	# so to avoid an annoying "sudo" step another docker container is created to get similar permissions
+	docker run \
+		--rm \
+		-v $(shell pwd)/pgosm-data:/app/output \
+		rustprooflabs/pgosm-flex \
+		bash -c "rm -rf /app/output/*"
+	rmdir pgosm-data|| echo "folder pgosm-data did not exist"
```
I would prefer to see removing the output directory as a dedicated optional step instead of running by default.  Always removing output directory requires downloading the files from Geofabrik every time. That's network traffic they don't need to be serving, and even with the small 17MB size of DC that can take a couple minutes when the download is running slow.

jacopofar replied:
Hm right, it's very annoying. I have a project with the same problem and tried to store the input file in `git-lfs` but ended up consuming the free tier in a few days.

Maybe having a separate make target only to download the file to run manually could be a compromise to have a completely reproducible process? IIRC the script downloads it only if necessary, so mounting the folders separately and deleting only the output one could be another option

rustprooflabs replied:
Oh -- Can we copy the files over from `tests/data/` into the `/pgosm-data` subdirectory to avoid the DL entirely?  That would mean using this method for testing would never test the download portion, I'm ok with that.  It would make testing this faster in general.

I had forgotten I committed the PBF/MD5 to make testing the imported data possible.

https://github.com/rustprooflabs/pgosm-flex/tree/main/tests/data

jacopofar replied:
Ah, I didn't notice. Now I changed the Makefile to copy these files in the target folder (and change the name into "latest"), the script already did a check so it doesn't download them when already present.

rustprooflabs commented on Makefile:
```diff
@@ -0,0 +1,32 @@
+.PHONY: docker-clean
+docker-clean:
+	# when docker runs as a demon the mounted folders have permissions and cannot be deleted by the current user
+	# so to avoid an annoying "sudo" step another docker container is created to get similar permissions
```
Yes, annoying!  I feel this should be fixable in the Dockerfile, I recall seeing another project setting UID/GID of files.  Opened #151

jacopofar replied:
Makes sense, should be the way to go

rustprooflabs reviewed (changes requested):
This will be helpful, thank you.

Running `make` runs only the clean command, I would expect that to also run `build-run-docker` by default.  A dedicated (optional) step for cleaning out the files would be great.

jacopofar commented:
> This will be helpful, thank you.
> 
> Running `make` runs only the clean command, I would expect that to also run `build-run-docker` by default. A dedicated (optional) step for cleaning out the files would be great.

yes, I assume that eventually the makefile will have different targets for the common tasks (build, test, coverage, push) so didn't assume the user has a default one in mind (I think by default if you invoke `make` without a target it runs the first in the file)

rustprooflabs commented:
Looking further at Docker permissions and volumes, none of the solutions seem elegant.  Ultimately I want to do better in Docker but that probably won't be a quick fix on my end.

I think we can leave the cleanup step you have as-is, and instead copy the file PBF/MD5 files from `tests/data/` to bypass the download.  Thoughts?

jacopofar commented:
Seems a good solution, but I miss how do the file arrive in that folder. Is it a manual download to be done beforehand?

rustprooflabs commented:
That file is part of the repo.  It was committed in #114 to support repeatable testing of the processed data.

jacopofar commented:
Ah, I didn't notice. Now I changed the makefile to copy these files in the target folder (and change the name into "latest"), the script already did a check so it doesn't download them when already present.

I noticed that these tests file are included in the normal Docker build, so I added a .dockerignore to exclude them.

The copy of the files in the target directory has to be done with `docker cp` instead of copying them directly for the same permission issues of the deletion

rustprooflabs commented:
Looks great, thank you!  I see the note about the extra copy for the missing bit in Python, I hope to get back to putting those missing pieces soon.
